static-site/src/components/ListResources/IndividualResource.jsx

```diff
@@ -0,0 +1,138 @@
+/* eslint-disable react/prop-types */
+import React, {useState, useCallback, useRef, useEffect} from 'react';
+import styled from 'styled-components';
+import { MdHistory } from "react-icons/md";
+
+
+/**
+ * These variables allow calculation of the width of <IndividualResource>,
+ * which we use for the column layout so that every <IndividualResource>
+ * is nicely aligned. There are other ways (e.g. querying the DOM) but
+ * this is simpler and seems to be working well.
+ */
+const [resourceFontSize, pxPerChar] = [16, 10];
+const iconWidth = 50; // including text
+const gapSize = 10;
+export const getMaxResourceWidth = (names) => {
+  const nameWidth = names.reduce((w, n) => {
+    const _w = n.length*pxPerChar;
+    return _w>w ? _w : w;
+  }, 200); // 200 is the minimum
+  return nameWidth + gapSize + iconWidth;
+}
+
+const ResourceLink = styled.a`
+  font-size: ${resourceFontSize}px;
+  font-family: monospace;
+  white-space: pre; /* don't collapse back-to-back spaces */
+  color: ${(props) => props.hovered ? '#31586c' : '#5097BA'} !important;
+  text-decoration: none !important;
+`;
+
+function Name({displayName, hovered, href, topOfColumn}) {
+  return (
+    <ResourceLink href={href} target="_blank" rel="noreferrer" hovered={hovered}>
+      {'• '}{(hovered||topOfColumn) ? displayName.hovered : displayName.default}
+    </ResourceLink>
+  )
+}
+
+const Container = styled.div`
+  padding: 3px;
+  overflow: hidden;
+  color: #4F4B50;
+`;
+
+const FlexRow = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: flex-start;
+  gap: ${gapSize}px;
+  align-items: center;
+`;
+
+export function TooltipWrapper({description, children}) {
+  return (
+    <div
+      data-tooltip-id="listResourcesTooltip"
+      data-tooltip-html={description}
+      data-tooltip-place="top">
+      {children}
+    </div>
+  )
+} 
+
+export function IconContainer({Icon, text, handleClick=undefined, color="#aaa"}) {
+  const iconProps = {size: "1.2em", color};
+  const hasOnClick = typeof handleClick === 'function';
+  return (
+    <div
+      style={{display: 'flex', color, alignItems: 'center', gap: '3px', cursor: hasOnClick?'pointer':'auto'}}
+      onClick={hasOnClick ? handleClick : ()=>{}}
+    >
+      <Icon {...iconProps}/>
+      {text}
+    </div>
+  )
+}
+
+
+/**
+ * 
+ * @param {*} param0 
+ * @returns 
+ */
+export const IndividualResource = ({data, isMobile}) => {
+  const ref = useRef(null);
+  const [topOfColumn, setTopOfColumn] = useState(false);
+  useEffect(() => {
+    /* The column CSS is great but doesn't allow us to know if an element is at
+    the top of its column, so we resort to JS */
+    if (ref.current.offsetTop===ref.current.parentNode.offsetTop) {
+      setTopOfColumn(true);
+    }
+  }, []);
```

Comment:
Seems like this does not always get reset when changing the sorting order. 
For pathogens with a lot of datasets, this can mess up the tree listing
 
![Screenshot 2024-04-01 at 5 02 15 PM](https://github.com/nextstrain/nextstrain.org/assets/40774762/b8cd9992-bac6-45e0-be82-a119d2214b10)

Reply:
Great observation! This resulted in two changes.

The bug is due to `<IndividualResource>` elements being tracked by React (via their unique `key`) and as we change sorting methods they re-render but aren't re-created and thus `useEffect` doesn't fire. There are a few ways to solve it but I made the key depend on the sorting method, as that's the approach we use to avoid this exact bug for window resizes, collapse state etc.

The second change I discovered while debugging this - I had been using `useEffect` within hooks `useFilterOptions` and `useSortAndFilter`. These effects were running after the components had been rendered (that's how `useEffect` works), which ultimately resulted in every `<IndividualResource>` rendering three times when it should only render once. I switched to `useMemo` which is the appropriate hook and fixes this. (Happy to talk through this more if anyone's interested.)

Reply:
>  I made the key depend on the sorting method, as that's the approach we use to avoid this exact bug for window resizes, collapse state etc.

Ah, thanks for explaining, I did not fully understand that updating the key will force a re-render, but that makes sense!